utils.go

```diff
@@ -14,14 +31,185 @@ type Storage struct {
 
 // String implements Storager.String
 func (s *Storage) String() string {
-	panic("implement me")
+	return fmt.Sprintf("Storager ftp {URL: %s, User: %s, WorkDir: %s}", s.url, s.user, s.workDir)
 }
 
 // NewStorager will create Storager only.
 func NewStorager(pairs ...types.Pair) (types.Storager, error) {
-	panic("implement me")
+	return newStorager(pairs...)
+}
+
+func newStorager(pairs ...types.Pair) (store *Storage, err error) {
+	defer func() {
+		if err != nil {
+			err = services.InitError{Op: "new_storager", Type: Type, Err: formatError(err), Pairs: pairs}
+		}
+	}()
+
+	store = &Storage{
+		connection: nil,
+		user:       "anonymous",
+		password:   "anonymous",
+		url:        "localhost:21",
+		workDir:    "/",
+	}
+
+	opt, err := parsePairStorageNew(pairs)
+	if err != nil {
+		return
+	}
+
+	if opt.HasEndpoint {
+		ep, err := endpoint.Parse(opt.Endpoint)
+		if err != nil {
+			return nil, err
+		}
+		var host string
+		var port int
+		switch ep.Protocol() {
+		case endpoint.ProtocolHTTP:
```

Comment:
Since we don't have `protocol ftp` supported now, it's ok for me to keep using `http` here.

It's better to add `protocol ftp` in go-endpoint, like [this issue](https://github.com/beyondstorage/go-endpoint/issues/9).

All these changes can be added in another PR.

Reply:
We can use protocol `tcp` instead.